src/main/java/nc/isi/http2kafka/Server.java

```diff
@@ -108,26 +118,37 @@ public Server(Properties producerProperties) {
 	}
 
 	public void handle(Request req, Response resp) {
+		statsd.incrementCounter("request");
 		if (LOG.isDebugEnabled()) {
 			LOG.debug("Request: " + req.getMethod() + " " + req.getPath());
 			LOG.trace(req);
 		}
 		try {
+			long startTime = System.nanoTime();
+
 			checkPath(req);
 			ensurePost(req);
 
 			String topic = extractTopic(req);
 			String key = extractKey(req);
 
+			statsd.incrementCounter("request." + topic);
 			sendBody(req, topic, key);
 
+			long endTime = System.nanoTime();
+			long duration = (endTime - startTime) / 1000000; // ms
```

Comment:
Probably, but [this](http://stackoverflow.com/questions/351565/system-currenttimemillis-vs-system-nanotime) has reasons.